### PR TITLE
Added workflow for release 

### PR DESCRIPTION
**Pull request by kshitiz-okahu**
## Proposed changes

The PR has workflow files to do following task
1. Add a prerelease workflow which creates release branch
2. Creates PRs for changelog updates.
3. Creates release notes.
4. Pushed package to pypi
 
## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "hatchling.build"
 
 [project]
 name = "monocle_apptrace"
-version = "0.2.0"
+version = "0.3.0"
```
Comment (oi-raanne) [inline pyproject.toml:7]: should we just do 0.2.1 ??

File: .github/release.md
Comment (oi-raanne) [inline .github/release.md:1]: this file is empty. Should we add the release procedure for the same?

**Review by oi-raanne — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (kshitiz-okahu) [reply pyproject.toml:7]: The versioning guide states that that for X.Y.Z , Z should be incremented if a patch is released with some fixes.
AN example is the opentelemetry SDK which mostly releases by incrementing Y if a new feature is releases and only updates Z when a patch(fix) is released

File: .github/release.md
Comment (kshitiz-okahu) [reply .github/release.md:1]: missed it, added now.

**Review by oi-raanne — APPROVED** [review]

**Pull request merged**